### ISSUE TITLE
Bot purchases and never sells

### ISSUE THREAD

**Issue by Jagermeister56** [yasinkuyu/binance-trader#66]
Bot purchases and never even tries to sell.
instead tries to keep buying more.

![screen shot 2018-01-08 at 12 48 18 am](https://user-images.githubusercontent.com/17785698/34660091-b4e7f7bc-f40d-11e7-9051-72d5162bd0ea.png)

**Comment by linhndd** [comment]
So i think you need buy some more BNB coin for trade fee. Or check your balance and make sure it enough for trade fee

**Comment by 001au** [comment]
This was an issue I had, and I resolved it by buying more BNB as well.

**Comment by Jagermeister56** [comment]
even with BNB this issue still seems to happen once every so often. it also continues to try buying over and over again. 
It is much more stable now, but still gets stuck every so often while being well above my profit parameters

**Comment by nealgentry** [comment]
Im Having the Same Issue With this the latest build.

**Comment by VasTShadowZz** [comment]
Same, i own 3 BNB coin

**Comment by linhndd** [comment]
yes, i see that is problem. Bot always buy coin and never set sell. I think we need turn back version. I think it run stable. We need waiting for fix this version and test it again

**Comment by jiqqaman** [comment]
Do you need to pay fees with BNB?

**Comment by VasTShadowZz** [comment]
@001au How much BNB do you own for the error not to accure?

**Comment by MIKEYCRYPTO** [comment]
I have BNB turned off in my settings because of the constant purchasing of coins . 

I had .5 eth and after running the bot, i had zillions of coins, and no eth left... Just kept buying and buying . Never selling even when the price dropped. 

Then I got the error of Insufficent Funds....
I didnt lose any $$ at all, my ETH balance just got spread around .

Is there any fix for the program to sell at some point?

**Comment by VasTShadowZz** [comment]
@MIKEYCRYPTO I stared small with just a test with the minimum, 0.0002BTC and it buys and sells like one or twice and then try to buy the 3rd time and then doesn't sell and trys to purchase again with the funds error.

**Comment by Jagermeister56** [comment]
now it's buying and selling, but sometimes it decides to buy at market value and gets stuck trying to sell it. 
Using BNB for fees. So far working great.. 
Maybe disable the buy at market value?

**Comment by linhndd** [comment]
@Jagermeister56  no, when I using BNB coin for fee, but it always buy and buy during my balance is lost of the BTC and show error.

**Comment by MIKEYCRYPTO** [comment]
Is the stop loss feature working?

Can someone post something that they are actually using that is working ok?

Currently im running

Python trader.py --quantity XXX --XRPETH --profit 1.5 --stoploss 4
--orderid 0 --wait_time 3 --test_mode false --loop 0

Is this a correct format?

**Comment by linhndd** [comment]
@MIKEYCRYPTO 
Eg:
python trader.py --quantity 10 --symbol XRPETH --profit 1.5

It enough

**Comment by MIKEYCRYPTO** [comment]
![20180109_235312](https://user-images.githubusercontent.com/35271112/34756577-e2db42fe-f599-11e7-8559-e88c4daf0ab5.jpg)
![20180109_235328](https://user-images.githubusercontent.com/35271112/34756581-ec290698-f599-11e7-894d-21b29d98d23c.jpg)
![20180109_235435 1](https://user-images.githubusercontent.com/35271112/34756585-f30f6a1a-f599-11e7-8c17-87415077d752.jpg)


All sorts of errors tonight after downloading the new build.

This is what i got after about 40 minutes running 3 windows.

**Comment by Jagermeister56** [comment]
@MIKEYCRYPTO MIN_NOTIONAL usually means you don't meet the minimum requirement for the purchase.
and all that code i have no idea, probably because you're trying to put a quantity under a full number
for example:
buying .3BTC with ETH
at least i always get a bunch of errors when i try that 
Also i don't believe the --stop_loss function works yet according to what it says in trader.py

**Comment by Jagermeister56** [comment]
@linhndd i'm also had the same problem, try to update the entire program 
mine is even worse now, it waits until the worst time and decides to buy it at market price because it can't keep up. THEN i'm stuck with a ton of currency crashing down. i wouldn't use this until he does a few more updates if you're worried about losing money.

**Comment by linhndd** [comment]
Yes, --stop_loss is the feature function. @yasinkuyu is prepare for next version. But I think he need fix problem is not sell and always buy coin

**Comment by linhndd** [comment]
Hi, i have a question ? After buy coin, how long it will be sell ? I see my bot running, it's always buy, and I waiting for sell, but it not work

**Comment by idoo** [comment]
@linhndd once parameters will achieve settings that you set

**Comment by gutch1** [comment]
Same thing. Am I doing something wrong or is it an issue? It will buy but never sells.

**Comment by yasinkuyu** [comment]
Devx branch made some improvements. Not tested. Error https://github.com/yasinkuyu/binance-trader/commits/devx

**Comment by yasinkuyu** [comment]
@linhndd Absolutely. --stop_loss is not ready yet.

**Comment by yasinkuyu** [comment]
@Jagermeister56  MIN_NOTIONAL error has been fixed.

**Comment by linhndd** [comment]
Great, I will be test it and feedback for you if have problem :)

**Comment by det-peralta** [comment]
Thank you for the updates. We really appreciate your effort. I just tested and:

![33](https://user-images.githubusercontent.com/35231379/34905962-5530916a-f85b-11e7-9f2c-f573778eb183.PNG)

**Comment by Memoi** [comment]
Hi,

Thanks for this great job Yasin. 

I have the same problem when I don't have enough BTC to buy an other coin, so may be in the code you can create an order sell if there is not enough BTC and just don't buy anymore until there is more BTC?

Thanks
